src/Platform/Microsoft.Testing.Platform/Builder/TestApplication.cs

```diff
@@ -311,7 +311,7 @@ private static ApplicationLoggingState CreateFileLoggerIfDiagnosticIsEnabled(
         }
 
         // Set the directory to the default test result directory
-        string directory = Path.Combine(Path.GetDirectoryName(testApplicationModuleInfo.GetCurrentTestApplicationFullPath())!, AggregatedConfiguration.DefaultTestResultFolderName);
+        string directory = Path.Combine(testApplicationModuleInfo.GetCurrentTestApplicationDirectory(), AggregatedConfiguration.DefaultTestResultFolderName);
```

Comment:
Refactoring the pattern of `Path.GetDirectoryName(GetCurrentTestApplicationFullPath())` to be using a new method `GetCurrentTestApplicationDirectory`. This is equivalent to the original logic, except that we don't throw on null path and fallback to `AppContext.BaseDirectory` instead.